mosdef_cassandra/core/moves.py

```diff
@@ -173,6 +181,77 @@ def __init__(self, ensemble, species_topologies):
                 self.prob_translate += self.prob_rotate
                 self.prob_rotate = 0.0
 
+    def add_restricted_insertions(
+        self, species_topologies, restricted_type, restricted_value
+    ):
+        """Add restricted insertions for specific species and boxes
+
+        Parameters
+        ----------
+        species_topologies : list
+            list of ``parmed.Structures``, with one species per element
+        restricted_type : list
+            list of restricted insertion types, with one species per element.
+        restricted_value : list
+            list of restricted insertion values, with one species per element.
+        """
+        if self.ensemble not in ["gcmc", "gemc", "gemc_npt"]:
+            raise ValueError(
+                "Restricted insertions are only valid for"
+                " 'gcmc', 'gemc', and 'gemc_npt' ensembles."
+            )
+        if len(restricted_type) != len(restricted_value):
+            raise ValueError(
+                "Length of 'restricted_type' and "
+                " 'restricted_value' must match."
+            )
+        for box in restricted_type:
+            if len(box) != len(species_topologies):
+                raise ValueError(
+                    "Length of 'species' and "
+                    " length of species list in 'restricted_type'"
+                    " must match."
+                )
+        for box in restricted_value:
+            if len(box) != len(species_topologies):
+                raise ValueError(
+                    "Length of 'species' and "
+                    " length of species list in 'restricted_value'"
+                    " must match."
+                )
```

Comment:
Perhaps these error messages can be made more verbose. If I do this:

```python
species_list = [methane]
moves = mc.Moves('gcmc', species_list)
moves.add_restricted_insertions(species_list, ['sphere'], [5.0])
```

The error message I get is:
```
ValueError: Length of 'species' and  length of species list in 'restricted_type' must match.
```

I'm not sure that points me towards fixing my problem.